packages/EasyApiToken/src/External/AwsCognitoJwtDriver.php

```diff
@@ -48,8 +56,16 @@ public function decode(string $token): object
             JWT::$leeway = $this->leeway;
         }
 
-        /** @var \stdClass $decodedToken */
-        $decodedToken = JWT::decode($token, $this->jwkFetcher->getJwks($this->userPoolConfig), $this->allowedAlgos);
+        $jwks = $this->jwkFetcher->getJwks($this->userPoolConfig);
+
+        if (self::isFirebaseJwtV6()) {
+            foreach ($jwks as $keyId => $key) {
+                $jwks[$keyId] = new Key($key, $this->defaultJwkAlgo);
```

Comment:
Are the JWKs already indexed by their keyId from the jwkFetcher? If not then it means that we have an array with each key twice in it, in raw format and as Key object.

Reply:
`$jwks` is `array<int|string, string>`. How can we have duplication?

Reply:
Because if the original `$jwks` isn't indexed by the keyIds and just by auto-increment integers then we will end up with having the original keys with auto-increments and the key objects indexed by their keyId

Reply:
`\EonX\EasyApiToken\External\AwsCognito\JwkFetcher::fetchKeysFromAws` returns an array of keys, with `kid` as an index.

I can't get what I do wrong.

Reply:
ok if that's the case then you're not doing anything wrong, I was asking for this confirmation